File: backend/src/main/java/com/yukon/logistics/model/dto/OrderResponse.java
package com.yukon.logistics.model.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;

@Builder
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class OrderResponse {
    private Long id;
    private Long cityFrom;
    private Long cityTo;
    private String cargoDescription;
    private Double cargoWeight;
    private TruckerResponse trucker;
    private CustomerResponse customer;
    private LocalDateTime created;
    private LocalDateTime modify;
    private LocalDateTime startDeliver;
    private LocalDateTime endDeliver;
    private Boolean completed;
}

File: frontend/src/app/models/order-response.ts
import { Trucker } from "./trucker";
import { User} from "./user"

export class OrderResponse {
    constructor(
        public id?: number,
        public cityFrom?: number,
        public cityTo?: number,
        public cargoDescription?: string,
        public cargoWeight?: number,
        public trucker?: Trucker | null | number,
        public customer?: User | null | number,
        public created?: string,
        public modified?: string,
        public startDeliver?: string,
        public endDeliver?: string,
        public complited?: boolean
    ) { }
}
